test/Microsoft.Extensions.Internal.Test/TypeNameHelperTest.cs

```diff
@@ -107,6 +108,25 @@ public void Returns_common_name_for_built_in_types()
             Assert.Equal("ulong", TypeNameHelper.GetTypeDisplayName(typeof(ulong)));
             Assert.Equal("ushort", TypeNameHelper.GetTypeDisplayName(typeof(ushort)));
         }
+
+        public static IEnumerable<object[]> GetTestDataForArrayTypes()
+        {
+            yield return new object[] { "int[]", typeof(int[]) };
+            yield return new object[] { "int[][]", typeof(int[][]) };
+            yield return new object[] { "int[,]", typeof(int[,]) };
+            yield return new object[] { "int[,,,]", typeof(int[,,,]) };
+            yield return new object[] { "int[,][,,]", typeof(int[,][,,]) };
+            yield return new object[] { "System.Collections.Generic.List<int[,][,,]>", typeof(List<int[,][,,]>) };
+            yield return new object[] { "List<int[,,][,]>[,][,,]", typeof(List<int[,,][,]>[,][,,]), false };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTestDataForArrayTypes))]
+        public void Can_pretty_print_array_name(string expected, Type type, bool fullName = true)
```

Comment:
Just write TheoryData inline instead of through MemberData. If that is not good for reading then lets just do what other tests in this class did.